stdlib/public/core/Sort.swift

```diff
@@ -696,6 +696,6 @@ extension UnsafeMutableBufferPointer {
     }
 
     // FIXME: Remove this, it works around rdar://problem/45044610
```

Comment:
This issue has perhaps been fixed, and maybe we can remove the precondition altogether.
An earlier PR to do that: https://github.com/apple/swift/pull/20714. (ping @natecook1000)
It may be worth trying to remove the precondition and then trying to run the benchmarks. If things go wrong, we can put it back.

Reply:
Since I don't have access to the `rdar` system, I can't really tell if it has been fixed or not. But if it's fixed I can, of course, remove the `precondition` and see how it goes.

Reply:
Sorry for not being clear; the radar is in an in-between state, so I'm not _certain_ it's fixed. Thus my "perhaps"!  
We should just try removing it.

Reply:
See https://bugs.swift.org/browse/SR-14750. This worked around an optimizer issue -- it looks like the optimizer may have been confused by the created array getting immediately discarded and that triggered a miscompilation. 

The radar is just a todo item. It does not contain information that helps figuring out whether this has been fixed -- the way to do that is to try removing the workaround and see if it triggers a regression.

(Btw, we recently added a new temporary allocation facility that may be a more idiomatic way to get temporary scratch space -- cc @natecook1000)

Reply:
...and it if doesn't work, the rdar link should be replaced with [SR-14750](https://bugs.swift.org/browse/SR-14750).